File: src/org/usfirst/frc/team4571/robot/commands/auto/SetElevatorTimed.java
package org.usfirst.frc.team4571.robot.commands.auto;

import edu.wpi.first.wpilibj.command.TimedCommand;
import org.usfirst.frc.team4571.robot.Robot;

public class SetElevatorTimed extends TimedCommand {
    private double speed;

    public SetElevatorTimed(double timeout, double speed) {
        super(timeout);
        requires(Robot.ELEVATOR);
        this.speed = speed;
    }

    protected void initialize() {}

    protected void execute() {
        Robot.ELEVATOR.setElevatorMotor(speed);
    }

    protected void end() {
        Robot.ELEVATOR.stopElevator();
    }

    protected void interrupted() {}
}

File: src/org/usfirst/frc/team4571/robot/commands/auto/SetElevatorTimed.kt
package org.usfirst.frc.team4571.robot.commands.auto

import edu.wpi.first.wpilibj.command.TimedCommand
import org.usfirst.frc.team4571.robot.subsystems.Elevator

class SetElevatorTimed(timeout: Double, private val speed: Double) : TimedCommand(timeout) {

    init {
        requires(Elevator)
    }

    override fun execute() = Elevator.setElevatorMotor(speed)

    override fun end() = Elevator.stopElevator()
}
